### DISCUSSION TITLE
ColumnTranformer error during predict

### DISCUSSION THREAD

**Discussion by MOTSci** [alan-turing-institute/sktime#4277] [Q&A]
Hi there,

I have a dataset set with a mix of numerical and categorical variables. And I would like to define a pipeline where: **(1)** the categorical variables remain the same, **(2)** for the first group of numerical variables I want to generate window features + differentiate them, **(3)** for the second group, I want to have only window features applied. 

To illustrate the error I'm getting, I load the _longley_ dataset and added a dummy variable, as shown below:
![image](https://user-images.githubusercontent.com/26097588/222266041-432d5aa0-e8c6-4165-9f96-94f01d32dcef.png)

The pipeline works fine until the fit command:

`# Define WindowSummarizer kwargs
kwargs = {
        "lag_feature": {
            "lag": [1],
            "mean": [[1, 3]]
        }
    }

numeric_features1 = ['GNPDEFL', 'GNP', 'ARMED']
numeric_features2 = ['UNEMP', 'POP']

numeric_transformer1 = make_pipeline(
    WindowSummarizer(n_jobs=1, **kwargs, target_cols=numeric_features1, truncate='bfill'),
    Differencer(lags=[1])
)

numeric_transformer2 = make_pipeline(
    WindowSummarizer(n_jobs=1, **kwargs, target_cols=numeric_features2, truncate='bfill'),
)

pipe_01 = ForecastingPipeline(
    steps=[
        ('preprocessing', 
         ColumnTransformer(
            transformers = [
                ('num1', numeric_transformer1, numeric_features1),
                ('num2', numeric_transformer2, numeric_features2),
            ],
            remainder = 'passthrough'
        )
        ),
        ('forecaster', NaiveForecaster(strategy='drift'))
    ]
)

pipe_01.fit(y_train, X_train)`

But, when I run `y_pred_01 = pipe_01.predict(fh=fh, X=X_test)`, I get the following error:

**ValueError: If passed as a pd.DataFrame, X must be a nested pd.DataFrame, with pd.Series or np.arrays inside cells.**

![image](https://user-images.githubusercontent.com/26097588/222267293-6bb25092-375d-461b-91d9-24f36f5fb2c2.png)

Has anyone faced the same issue before? Could you help me with this?

Thank you.

**Comment by MOTSci** [comment]
Although issue #4056 solves part of my issues, the solution proposed doesn't work with `WindowSummarizer`.  Now, I'm getting  **ValueError: Columns must be same length as key** on the following piece of code:

`
num_features = ['GNPDEFL', 'GNP', 'UNEMP', 'ARMED', 'POP']

kwargs = {
    "lag_feature": {
        "lag": [1],
        "mean": [[1, 3]]
   }
}

transformer_01 = (
    (
        ColumnwiseTransformer(WindowSummarizer(**kwargs) * Differencer(), num_features)
    )
)

X_transformed_01 = transformer_01.fit_transform(X)

X_transformed_01
`



It's interesting that the code works when I have only `"lag": [1]` on the `kwargs`

Any advice?

**Comment by fkiraly** [comment]
Thanks for reporting, @motuchsznajder! Sounds like a bug.
Could you kindly open a bug report on the issue tracker, with a single block of complete code including how you generate the dummy data based on longley? Much appreciated, that allows faster diagnosis and fixing.
